lgc/test/s.buffer.load.imm.lgc

```diff
@@ -0,0 +1,52 @@
+; Test that @llvm.amdgcn.s.buffer.load.imm.i32 with non-zero immediate offset gets generated.
+; Offset 10 is equal to offset (1, 2) inside %llpc.array.element.1: 2 * 4 + 2 * 1.
+
+; RUN: lgc -mcpu=gfx1010 -enable-opaque-pointers -print-after=lgc-patch-buffer-op -o - - <%s 2>&1 | FileCheck --check-prefixes=CHECK %s
```

Comment:
Is it possible to convert this test to use `-passes=lgc-patch-buffer-op` so that it runs only the target pass? And then use update_test_checks.py (from a recent-ish LLVM update where it supports the `--tool` and `--tool-binary` options) to generate the CHECK lines?

Reply:
Sure, will do.

Reply:
It didn't work for me when I tried to convert the test. Unless there is something trivial I'm missing, I suspect the pipeline state is not properly serialized, so it returns null.

```
Program received signal SIGSEGV, Segmentation fault.
lgc::PipelineStateWrapperResult::getPipelineState (this=0x0) at /work/gpuopen/drivers/llpc/lgc/include/lgc/state/PipelineState.h:524
524	  PipelineState *getPipelineState() { return m_pipelineState; }
(gdb) p m_pipelineState
Cannot access memory at address 0x0
```